packages/host/tests/acceptance/operator-mode-test.ts

```diff
@@ -0,0 +1,317 @@
+      `,
+      'Pet/mango.json': {
+        data: {
+          type: 'card',
+          id: `${testRealmURL}Pet/mango`,
+          attributes: {
+            name: 'Mango',
+          },
+          meta: {
+            adoptsFrom: {
+              module: `${testRealmURL}pet`,
+              name: 'Pet',
+            },
+          },
+        },
+      },
+
+      'Person/fadhlan.json': {
+        data: {
+          type: 'card',
+          id: `${testRealmURL}Person/fadhlan`,
+          attributes: {
+            firstName: 'Fadhlan',
+            address: {
+              city: 'Bandung',
+              country: 'Indonesia',
+            },
+          },
+          relationships: {
+            pet: {
+              links: {
+                self: `${testRealmURL}Pet/mango`,
+              },
+            },
+          },
+          meta: {
+            adoptsFrom: {
+              module: `${testRealmURL}person`,
+              name: 'Person',
+            },
+          },
+        },
+      },
+      'grid.json': {
+        data: {
+          type: 'card',
+          attributes: {},
+          meta: {
+            adoptsFrom: {
+              module: 'https://cardstack.com/base/cards-grid',
+              name: 'CardsGrid',
+            },
+          },
+        },
+      },
+      'index.json': {
+        data: {
+          type: 'card',
+          attributes: {},
+          meta: {
+            adoptsFrom: {
+              module: 'https://cardstack.com/base/cards-grid',
+              name: 'CardsGrid',
+            },
+          },
+        },
+      },
+    });
+
+    realm = await TestRealm.createWithAdapter(adapter, this.owner, {
+      isAcceptanceTest: true,
+    });
+
+    let loader = (this.owner.lookup('service:loader-service') as LoaderService)
+      .loader;
+    loader.registerURLHandler(new URL(realm.url), realm.handle.bind(realm));
+    await realm.ready;
+  });
+
+  test('visiting index card and entering operator mode', async function (assert) {
+    await visit('/');
+
+    assert.strictEqual(currentURL(), '/');
+
+    // Enter operator mode
+    await triggerEvent(document.body, 'keydown', {
+      code: 'Key.',
+      key: '.',
+      ctrlKey: true,
+    });
+
+    assert.dom('.operator-mode-card-stack').exists();
+    assert.dom('[data-test-stack-card-index="0"]').exists(); // Index card opens in the stack
+
+    // In the URL, operatorModeEnabled is set to true and operatorModeState is set to the current stack
+    assert.strictEqual(
+      currentURL(),
+      `/?operatorModeEnabled=true&operatorModeState=${encodeURIComponent(
+        JSON.stringify({
+          stacks: [
+            {
+              items: [
+                {
+                  card: { id: 'http://test-realm/test/index' },
+                  format: 'isolated',
+                },
+              ],
+            },
+          ],
+        })
+      )}`
+    );
+  });
+
+  test('restoring the stack from query param', async function (assert) {
```

Comment:
should we also include a test that restores a format = 'edit' card as well?

Reply:
Makes a lot of sense - I added a test case for that.